systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaUtils.java

```diff
@@ -25,21 +26,29 @@
 
     private KafkaUtils() {}
 
-    public static void waitUntilKafkaCRIsReady(String clusterName) {
+    public static void waitUntilKafkaStatusConditionIsReady(String clusterName) {
         LOGGER.info("Waiting till Kafka CR will be ready");
+
         TestUtils.waitFor("Waiting for Kafka resource status is ready", Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_TIMEOUT,
-            () ->   Crds.kafkaOperation(kubeClient().getClient()).inNamespace(kubeClient().getNamespace()).withName(clusterName).get().getStatus().getConditions().get(0).getType().equals("Ready") &&
-                Crds.kafkaOperation(kubeClient().getClient()).inNamespace(kubeClient().getNamespace()).withName(clusterName).get().getStatus().getConditions().get(0).getStatus().equals("True")
+            () ->  {
+                Condition condition = Crds.kafkaOperation(kubeClient().getClient()).inNamespace(kubeClient().getNamespace()).withName(clusterName).get().getStatus().getConditions().get(0);
+                return condition.getType().equals("Ready") && condition.getStatus().equals("True");
+            }
         );
         LOGGER.info("Kafka CR will be ready");
     }
 
-    public static void waitUntilKafkaStatusConditionIsPresent(String clusterName) {
-        LOGGER.info("Waiting till kafka resource status is present");
-        TestUtils.waitFor("Waiting for Kafka resource status is ready", Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_TIMEOUT,
-            () ->  Crds.kafkaOperation(kubeClient().getClient()).inNamespace(kubeClient().getNamespace()).withName(clusterName).get().getStatus().getConditions().get(0) != null
+    public static void waitUntilKafkaStatusConditionIsNotReady(String clusterName, String message) {
+        LOGGER.info("Waiting till kafka resource status is not ready");
+
+        TestUtils.waitFor("Waiting for Kafka resource status is not ready", Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_TIMEOUT,
+            () ->  {
+                Condition condition = Crds.kafkaOperation(kubeClient().getClient()).inNamespace(kubeClient().getNamespace()).withName(clusterName).get().getStatus().getConditions().get(0);
+                LOGGER.info("Type:{}, Status:{}, Message:{}", condition.getType(), condition.getStatus(), condition.getMessage());
+                return condition.getType().equals("NotReady") && condition.getStatus().equals("True") && condition.getMessage().contains(message);
+            }
         );
-        LOGGER.info("Kafka resource status is present");
+        LOGGER.info("Kafka resource status is not ready");
```

Comment:
A nit pick, maybe you can add the message to the log message here and to the message on line 44. But that is just a detail.